source/PreferencesPanel.cpp

```diff
@@ -93,6 +93,17 @@ namespace {
 PreferencesPanel::PreferencesPanel()
 	: editing(-1), selected(0), hover(-1)
 {
+	page = Preferences::GetPreviousPage();
+	if(!page)
+		page = 'c';
+	currentControlsPage = 0;
+	currentSettingsPage = 0;
+	int pagination = Preferences::GetPreviousPagination();
+	if(page == 'c')
+		currentControlsPage = pagination;
+	else if(page == 's')
+		currentSettingsPage = pagination;
```

Comment:
You should check if these pages actually exist (less than CONTROLS_PAGE_COUNT or SETTINGS_PAGE_COUNT). Might as well use std::clamp with a minimum of 0.